### PR TITLE
lots of little fixes

### PR DESCRIPTION
**Pull request by bqpd**
@whoburg, ready for review

### CODE REVIEW

**Comment by bqpd** [comment]
retest this please (dependency test seems to be hanging...)

**Comment by bqpd** [comment]
@whoburg :smile:

File: gpkit/constraints/tight.py [hunk 1]
```diff
@@ -20,8 +21,11 @@ def process_result(self, result):
         for constraint in self.flat():
             if isinstance(constraint, (PosynomialInequality,
                                        SignomialInequality)):
-                leftsubbed = constraint.left.sub(variables).value
-                rightsubbed = constraint.right.sub(variables).value
+                with SignomialsEnabled():
```
Comment (bqpd) [inline gpkit/constraints/tight.py]: might be unnecessary depending on #745

File: gpkit/nomials/nomial_math.py [hunk 2]
```diff
@@ -522,10 +521,10 @@ def as_posyslt1(self):
...
         for posy in posys:
             _, exps, cs, _ = substitution(posy, self.substitutions)
-            # remove any cs that are just nans and/or 0s
-            nans = np.isnan(cs)
-            if np.all(nans) or np.all(cs[~nans] == 0):
-                continue  # skip nan'd or 0'd constraint
+            # remove any cs that are trivial (0s) or impossible (inf, nan)
+            finite = np.isfinite(cs)
+            if not np.any(finite) or np.all(cs[finite] == 0):
+                continue  # skip the constraint
```
Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Don't do this.

**Comment by mayork** [comment]
Tried to do a signomial equality in the last commit (8264dd7) here..https://github.com/hoburg/gpkit-models/pull/73

got this error

'''Traceback (most recent call last):
  File "C:\Users\mayork\Documents\GitHub\gpkit-models\gpkitmodels\Commercial Sizing\Commerical_Flight_Profile.py", line 384, in <module>
    m = CommercialAircraft()
  File "C:\Users\mayork\Documents\GitHub\gpkit-models\gpkitmodels\Commercial Sizing\Commerical_Flight_Profile.py", line 359, in __init__
    climb1 = Climb1()
  File "C:\Users\mayork\Documents\GitHub\gpkit-models\gpkitmodels\Commercial Sizing\Commerical_Flight_Profile.py", line 251, in __init__
    Model.__init__(self, None, constraints, **kwargs)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\model.py", line 65, in __init__
    self._add_modelname_tovars(self.name, self.num)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\model.py", line 100, in _add_modelname_tovars
    self.subinplace(add_model_subs)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\costed.py", line 25, in subinplace
    ConstraintSet.subinplace(self, subs, value)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\set.py", line 128, in subinplace
    constraint.subinplace(subs, value)
  File "c:\users\mayork\documents\github\gpkit\gpkit\constraints\set.py", line 128, in subinplace
    constraint.subinplace(subs, value)
AttributeError: 'bool' object has no attribute 'subinplace'''

I checked out the branch and installed so I'm not sure exactly what's going on

**Comment by bqpd** [comment]
@mayork, you can't use `==` to make SignomialEquality constraints, which looks like it might have been what happened? Instead use `SignomialEquality(left, right)` for now...

**Comment by mayork** [comment]
Yeah that's what I was doing. That my fault, should've verified that syntax

**Comment by bqpd** [comment]
Is it working, then? :D

**Comment by mayork** [comment]
@bqpd yeah its awesome! Code actually ran faster with the signomial equality constraint. Without it was 5 GP solves and 1.14 seconds, with it 5 GP solves and only 1.02 seconds....I'm going to have to dig into that paper some more this is pretty interesting

**Pull request merged**